Rework throw statement description and examples (#27297)
mdn/content: files/en-us/web/javascript/reference/statements/return/index.md

return expression;
```

- `expression` {{optional_inline}}
  - : The expression whose value is to be returned. If omitted, `undefined` is returned.

## Description

The `return` statement can only be used within function bodies. When a `return` statement is used in a function body, the execution of the function is stopped. The `return` statement has different effects when placed in different functions:

- In a plain function, the call to that function evaluates to the return value.
- In an async function, the produced promise is resolved with the returned value.
- In a generator function, the produced generator object's `next()` method returns `{ done: true, value: returnedValue }`.
- In an async generator function, the produced async generator object's `next()` method returns a promise fulfilled with `{ done: true, value: returnedValue }`.

If a `return` statement is executed within a {{jsxref("Statements/try...catch", "try")}} block, its `finally` block, if present, is first executed, before the value is actually returned.

### Automatic semicolon insertion

> The syntax forbids any line terminator between the `return` keyword and the expression, therefore introducing [automatic semicolon insertion (ASI)](/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#automatic_semicolon_insertion) hazards.

Reviewer: ```suggestion
The syntax forbids line terminators between the `return` keyword and an expression to be returned.
```

Then below instead if "is transformed by ASI into" you might have

> is transformed by [automatic semicolon insertion (ASI)](/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#automatic_semicolon_insertion) into.
